### DISCUSSION TITLE
Google Mobile ADS Sdk

### DISCUSSION THREAD

**Discussion by devGuerra** [invertase/react-native-firebase#5052] [Q&A] [answered]
Hi,

Recently i try use  [this lib](https://github.com/ammarahm-ed/react-native-admob-native-ads) to create native ads in my app, but case i try use Admob firebase lib for create my ads banner and interstitial i receive this error, any can help me ? 

<img width="718" alt="Captura de Tela 2021-03-18 às 18 13 36" src="https://user-images.githubusercontent.com/6579736/111699520-0128ab80-8817-11eb-9d22-edea1160fe64.png">

my deps

> "dependencies": {
    "@react-native-async-storage/async-storage": "^1.13.4",
    "@react-native-community/masked-view": "^0.1.10",
    "@react-native-firebase/admob": "^11.1.2",
    "@react-native-firebase/analytics": "^11.1.2",
    "@react-native-firebase/app": "^11.1.2",
    "@react-native-picker/picker": "^1.9.11",
    "@react-navigation/bottom-tabs": "^5.11.7",
    "@react-navigation/native": "^5.8.9",
    "@react-navigation/stack": "^5.12.6",
    "@unform/core": "2.1.5",
    "@unform/mobile": "2.1.5",
    "axios": "^0.21.0",
    "date-fns": "^2.17.0",
    "react": "^17.0.1",
    "react-native": "0.63.3",
    "react-native-admob-native-ads": "^0.4.0",
    "react-native-device-info": "^8.0.6",
    "react-native-flipper": "^0.71.0",
    "react-native-gesture-handler": "^1.8.0",
    "react-native-iap": "^5.2.7",
    "react-native-in-app-review": "^3.1.0",
    "react-native-iphone-x-helper": "^1.3.1",
    "react-native-picker-select": "^8.0.4",
    "react-native-reanimated": "^1.13.1",
    "react-native-safe-area-context": "^3.1.8",
    "react-native-screens": "^2.14.0",
    "react-native-share": "^5.1.1",
    "react-native-svg": "^12.1.0",
    "react-native-svg-transformer": "^0.14.3",
    "react-native-vector-icons": "^8.1.0",
    "styled-components": "^5.2.1",
    "unstated-next": "^1.1.0",
    "yup": "^0.29.3"
  },

**Comment by mikehardy** [comment, marked as answer]
So you have two admob dependencies at once, and they have conflicting transitive dependencies?
I think you'll either need to pick one, or it's possible you will always struggle to match up the versions expected by both

This one in particular is not in react-native-firebase control, it's from upstream firebase-ios-sdk https://github.com/CocoaPods/Specs/blob/455406ece920a20cad26fc8551890e5db3823aeb/Specs/0/3/5/Firebase/7.8.1/Firebase.podspec.json#L103 / https://github.com/firebase/firebase-ios-sdk/blob/721a7de3bd573fcbc50d9112d66d6136918ac30f/Firebase.podspec#L73

Unfortunately you can't just use both and force one to accept the other's version in your Podfile as there are a number of breaking changes in the shift to Google-Mobile-Ads-SDK v8 https://developers.google.com/admob/ios/migration

So you'll need to ask on firebase-ios-sdk (as a question in their github repository) if they have any intention of forward-porting to v8.
